scripts/ci/extract_allure_failures.py crashes with UnicodeEncodeError on Windows job logs (emoji in output)

## Summary

Found while diagnosing nightly issue #3987 (Local E2E Tests). The `post-test-report` composite action runs `python3 scripts/ci/extract_allure_failures.py` (`scripts/ci/extract_allure_failures.py:490`) to print a `❌ Failed and broken test methods` table into the job's step summary/log whenever a Windows job's Allure report shows failures or broken tests. On `windows-latest` runners, Python's default stdout encoding is `cp1252`, not UTF-8, and `print(output)` crashes:

```
Traceback (most recent call last):
  File "D:\a\SHAFT_ENGINE\SHAFT_ENGINE\scripts\ci\extract_allure_failures.py", line 495, in <module>
    raise SystemExit(main())
  File "D:\a\SHAFT_ENGINE\SHAFT_ENGINE\scripts\ci\extract_allure_failures.py", line 490, in main
    print(output)
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u274c' in position 352: character maps to <undefined>
```

Reproduced identically in both `Windows_Chrome_Local` and `Windows_Edge_Local` in run [30143597510](https://github.com/ShaftHQ/SHAFT_ENGINE/actions/runs/30143597510) (and presumably every prior nightly run where a Windows local job had any failed/broken test — this makes it a long-standing latent bug, only ever silent because it needs a failure to trigger). The script's own crash is swallowed (`|| true` at the call site, `.github/actions/post-test-report`), so the job still fails correctly on the real `##[error]Allure report shows N failed and M broken` check — but the actual per-test failure table (test name, error message, source file) that the same script successfully prints on Ubuntu/macOS is completely missing from Windows job logs and step summaries, which cost real diagnosis time while investigating #3987 (had to fall back to grepping raw Surefire/mvn output instead).

## Proposed minimal fix

Force UTF-8 stdout in the script, e.g. `sys.stdout.reconfigure(encoding="utf-8")` near the top of `scripts/ci/extract_allure_failures.py`'s `main()`, or set `PYTHONUTF8=1`/`PYTHONIOENCODING=utf-8` in the `post-test-report` composite action's environment before invoking `python3`. This is Windows-only; Ubuntu/macOS runners already default to a UTF-8 locale.

Filed while diagnosing #3987, not fixed — read-only diagnosis pass, no code changes made.

--- a/scripts/ci/extract_allure_failures.py
+++ b/scripts/ci/extract_allure_failures.py
@@ -14,6 +14,7 @@ import argparse
 import base64
 import json
 import re
+import sys
 import textwrap
 import xml.etree.ElementTree as ET
 import zipfile
@@ -474,6 +475,15 @@ def parse_args() -> argparse.Namespace:
 
 
 def main() -> int:
+    # Windows CI runners (and any non-UTF-8 console/pipe) default sys.stdout to
+    # a codec like cp1252 that cannot encode the emoji SHAFT's own assertion
+    # output contains, crashing this script's print() with UnicodeEncodeError
+    # (#4007). Force UTF-8 so the emoji print correctly instead of being
+    # stripped or crashing. Guarded because not every stdout stand-in (e.g.
+    # io.StringIO in tests) implements reconfigure().
+    if hasattr(sys.stdout, "reconfigure"):
+        sys.stdout.reconfigure(encoding="utf-8")
+
     args = parse_args()
     allure_failures = extract_failures(args.paths)
     surefire_failures = extract_surefire_failures(args.surefire_reports)

--- a/tests/scripts/test_extract_allure_failures.py
+++ b/tests/scripts/test_extract_allure_failures.py
@@ -362,6 +362,38 @@ class ExtractAllureFailuresTests(unittest.TestCase):
         self.assertIn("visible exception", markdown)
         self.assertNotIn("hidden fixture should not be counted", markdown)
 
+    def test_prints_emoji_bearing_failure_reason_without_crashing_on_non_utf8_stdout(self):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            root = Path(temp_dir)
+            (root / "failed-result.json").write_text(
+                json.dumps(
+                    {
+                        "status": "failed",
+                        "fullName": "pkg.EmojiTest.fails",
+                        "statusDetails": {
+                            "message": "❌ assertion failed \U0001f50d see \U0001f4cd trace",
+                        },
+                    }
+                ),
+                encoding="utf-8",
+            )
+
+            buffer = io.BytesIO()
+            # Simulate a Windows CI runner whose stdout code page is cp1252, not
+            # UTF-8 -- this is what sys.stdout looks like on windows-latest when
+            # python's stdout is piped (e.g. `python3 ... | tee`), reproducing #4007.
+            # Force the encoding explicitly rather than relying on the dev shell's
+            # ambient (already-UTF-8) console.
+            non_utf8_stdout = io.TextIOWrapper(buffer, encoding="cp1252", errors="strict")
+            with patch("sys.argv", ["extract_allure_failures.py", str(root)]), patch("sys.stdout", non_utf8_stdout):
+                exit_code = main()
+                non_utf8_stdout.flush()
+
+        self.assertEqual(exit_code, 1)
+        printed = buffer.getvalue().decode("utf-8")
+        self.assertIn("❌", printed)
+        self.assertIn("pkg.EmojiTest.fails", printed)
+
     def test_ignores_hidden_allure_failures_from_generated_report_json(self):
         with tempfile.TemporaryDirectory() as temp_dir:
             report_data = Path(temp_dir) / "data" / "test-results"

